app/components/emoji_picker/emoji_picker.js

```diff
@@ -161,23 +165,22 @@ export default class EmojiPicker extends PureComponent {
     };
 
     changeSearchTerm = (text) => {
-        const nextState = {
-            searchTerm: text,
-        };
+        const searchTerm = filterEmojiSearchInput(text);
+        const nextState = {searchTerm};
 
-        if (!text) {
+        if (!searchTerm) {
             nextState.currentSectionIndex = 0;
         }
 
         this.setState(nextState);
 
         clearTimeout(this.searchTermTimeout);
-        const timeout = text ? 100 : 0;
+        const timeout = searchTerm ? 100 : 0;
         this.searchTermTimeout = setTimeout(async () => {
             if (isMinimumServerVersion(this.props.serverVersion, 4, 7)) {
-                await this.props.actions.searchCustomEmojis(text);
+                await this.props.actions.searchCustomEmojis(searchTerm);
             }
-            const filteredEmojis = this.searchEmojis(text);
+            const filteredEmojis = this.searchEmojis(searchTerm);
```

Comment:
Should we only search whenever the filtered search term is not empty string?  Might use trim as necessary.